src/test/java/com/onaio/steps/handler/ExceptionHandlerTest.java

```diff
@@ -57,13 +65,17 @@ public void testHandleMultipleExceptionScenarios() {
         verify(handler, times(1)).postDelayed(acRunnable.capture(), eq(1000L));
 
         acRunnable.getValue().run();
-        verify(menuHandler, times(1)).open();
+        verify(resolvableException, times(1)).tryToResolve();
 
         exceptionHandler.handle(mock(NullPointerException.class));
         verifyAlertError(R.string.something_went_wrong_try_again);
+
+        verify(callback, times(3)).onDismiss(any(Exception.class), anyInt());
     }
 
     private void verifyAlertError(int message) {
-        verify(customDialog, times(1)).notify(activity, CustomDialog.EmptyListener, R.string.error_title, message);
+        ArgumentCaptor<DialogInterface.OnClickListener> arOnClickListener = ArgumentCaptor.forClass(DialogInterface.OnClickListener.class);
+        verify(customDialog, times(1)).notify(eq(activity), arOnClickListener.capture(), eq(R.string.error_title), eq(message));
```

Comment:
Move line 78 and 79 to the specific method where this is used. Change this method return type to the `ArgumentCaptor` so that you can use it in that method.

I noticed that you use those lines only once but this method is invoked in other tests where this is not required

Reply:
logic updated no more need to use `ArgumentCaptor`7